driver: clock_control: RCC_BDCR_LSEDRV_Pos undefined using STM32L0

### Describe the bug

Currently STM32 LSE driving capability is set using `RCC_BDCR_LSEDRV_Pos`, however STM32L0 defines `RCC_CSR_LSEDRV_Pos` instead of the used one.

https://github.com/zephyrproject-rtos/zephyr/blob/b70f7d28c623ee3e66737441b962de1a4cfda389/drivers/clock_control/clock_stm32_ll_common.c#L1015-L1018

### Regression

- [ ] This is a regression.

### Steps to reproduce

Set LSE clock driving capability in overlay file using for instance `nucleo_l053r8` board:

```dts
&clk_lse {
	driving-capability = <3>;
	status = "okay";
};
```

### Relevant log output

```shell
FAILED: zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_stm32_ll_common.c.obj

...

/opt/toolchains/zephyr/drivers/clock_control/clock_stm32_ll_common.c: In function 'set_up_fixed_clock_sources':
/opt/toolchains/zephyr/drivers/clock_control/clock_stm32_ll_common.c:677:68: error: 'RCC_BDCR_LSEDRV_Pos' undeclared (first use in this function); did you mean 'RCC_CSR_LSEDRV_Pos'?
  677 |                 LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
      |                                                                    ^~~~~~~~~~~~~~~~~~~
      |                                                                    RCC_CSR_LSEDRV_Pos
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

Linux
Zephyr v3.7.1 (but looking at sousces main branch is buggy too)
SDK v0.16.9
App `samples/drivers/uart/echo_bot` using board `nucleo_l053r8`

### Additional Context

An app-side workaround is adding to `CMakeLists.txt`:

```cmake
zephyr_compile_definitions(RCC_BDCR_LSEDRV_Pos=RCC_CSR_LSEDRV_Pos)
```

--- a/drivers/clock_control/clock_stm32_ll_common.c
+++ b/drivers/clock_control/clock_stm32_ll_common.c
@@ -1013,8 +1013,20 @@ static void set_up_fixed_clock_sources(void)
 		stm32_backup_domain_enable_access();
 
 #if STM32_LSE_DRIVING
+/*
+ * Most series have LSEDRV field in RCC_BDCR register,
+ * but a few series have it in a different one yet are
+ * handled by this driver. Pick proper register name:
+ */
+#define LSE_DRIVING_SHIFT					\
+	COND_CODE_1(IS_ENABLED(CONFIG_SOC_SERIES_STM32C0X),	\
+		(RCC_CSR1_LSEDRV_Pos),				\
+	(COND_CODE_1(IS_ENABLED(CONFIG_SOC_SERIES_STM32L0X),	\
+		(RCC_CSR_LSEDRV_Pos),				\
+		(RCC_BDCR_LSEDRV_Pos))))
+
 		/* Configure driving capability */
-		LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
+		LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << LSE_DRIVING_SHIFT);
 #endif
 
 		if (IS_ENABLED(STM32_LSE_BYPASS)) {

